packages/tangle/messagefactory_test.go

```diff
@@ -146,38 +160,203 @@ func TestMessageFactory_POW(t *testing.T) {
 	assert.NoError(t, err)
 }
 
-func TestWorkerFunc_PayloadSize(t *testing.T) {
-	// TODO
-	/*
-		testTangle := newTestTangle()
-		defer testTangle.Shutdown()
-
-		msgFactory := NewMessageFactory(
-			testTangle,
-			TipSelectorFunc(func(p payload.Payload, countStrongParents, countWeakParents int) (strongParents, weakParents MessageIDs, err error) {
-				result := make(MessageIDs, 0, MaxParentsCount)
-				for i := 0; i < MaxParentsCount; i++ {
-					b := make([]byte, MessageIDLength)
-					_, _ = rand.Read(b)
-					randID, _, _ := MessageIDFromBytes(b)
-					result = append(result, randID)
-				}
-				return result, MessageIDs{}, nil
-			}),
-		)
-		defer msgFactory.Shutdown()
-
-		// issue message with max allowed payload size
-		// dataPayload headers: type|32bit + size|32bit
-		data := make([]byte, payload.MaxSize-4-4)
-		msg, err := msgFactory.IssuePayload(payload.NewGenericDataPayload(data))
-		require.NoError(t, err)
-		assert.Truef(t, MaxMessageSize == len(msg.Bytes()), "message size should be exactly %d bytes but is %d", MaxMessageSize, len(msg.Bytes()))
-
-		// issue message bigger than max allowed payload size
-		data = make([]byte, payload.MaxSize)
-		msg, err = msgFactory.IssuePayload(payload.NewGenericDataPayload(data))
-		require.Error(t, err)
-		assert.Nil(t, msg)
-	*/
+func TestMessageFactory_PrepareLikedReferences_1(t *testing.T) {
+	tangle := NewTestTangle()
+	defer tangle.Shutdown()
+
+	testFramework := NewMessageTestFramework(
+		tangle,
+		WithGenesisOutput("O1", 500),
+		WithGenesisOutput("O2", 500),
+	)
+
+	tangle.Setup()
+
+	tangle.Events.Error.Attach(events.NewClosure(func(err error) {
+		t.Logf("Error fired: %v", err)
+	}))
+
+	branches := make(map[string]ledgerstate.BranchID)
+
+	// Message 1
+	testFramework.CreateMessage("1", WithStrongParents("Genesis"), WithInputs("O1"), WithOutput("O3", 500))
+	testFramework.IssueMessages("1").WaitMessagesBooked()
+
+	// Message 2
+	testFramework.CreateMessage("2", WithStrongParents("Genesis"), WithInputs("O2"), WithOutput("O5", 500))
+	testFramework.IssueMessages("2").WaitMessagesBooked()
+
+	// Message 3
+	testFramework.CreateMessage("3", WithStrongParents("Genesis"), WithInputs("O2", "O1"), WithOutput("O4", 1000))
+	testFramework.IssueMessages("3").WaitMessagesBooked()
```

Comment:
You could issue all 3 messages at once since you do not check intermediate states. This would make it shorter and probably more readable.

```suggestion
	testFramework.IssueMessages("1", "2", "3").WaitMessagesBooked()
```